src/internal/connector/sharepoint/restore.go

```diff
@@ -271,6 +274,81 @@ func RestoreCollection(
 	}
 }
 
+// RestorePageCollection handles restoration of an individual site page collection.
+// returns:
+// - the collection's item and byte count metrics
+// - the context cancellation station. True iff context is canceled.
+func RestorePageCollection(
+	ctx context.Context,
+	service graph.Servicer,
+	dc data.Collection,
+	restoreContainerName string,
+	deets *details.Builder,
+	errUpdater func(string, error),
+) (support.CollectionMetrics, bool) {
+	ctx, end := D.Span(ctx, "gc:sharepoint:restorePageCollection", D.Label("path", dc.FullPath()))
+	defer end()
+
+	var (
+		metrics   = support.CollectionMetrics{}
+		directory = dc.FullPath()
+	)
+
+	trace.Log(ctx, "gc:sharepoint:restorePageCollection", directory.String())
+	siteID := directory.ResourceOwner()
+
+	// Restore items from collection
+	items := dc.Items()
+
+	for {
+		select {
+		case <-ctx.Done():
+			errUpdater("context canceled", ctx.Err())
+			return metrics, true
+
+		case itemData, ok := <-items:
+			if !ok {
+				return metrics, false
+			}
+			metrics.Objects++
+
+			itemInfo, err := restoreSitePage(
+				ctx,
+				service,
+				itemData,
+				siteID,
+				restoreContainerName,
+			)
+			if err != nil {
+				errUpdater(itemData.UUID(), err)
+				continue
+			}
+
+			metrics.TotalBytes += itemInfo.SharePoint.Size
+
+			itemPath, err := dc.FullPath().Append(itemData.UUID(), true)
+			if err != nil {
+				logger.Ctx(ctx).DPanicw("transforming item to full path", "error", err)
```

Comment:
Avoid `Dpanic` logging, please.
```suggestion
				logger.Ctx(ctx).Errorw("transforming item to full path", "error", err)
```